scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,191 @@
+    parser.add_argument('-c', '--config', 
+                        help='path to CCPP prebuild configuration file', required=True)
+    parser.add_argument('-v', '--variable', help='variable to track through CCPP suite', 
+                        required=True)
+    parser.add_argument('--debug', action='store_true', help='enable debugging output', 
+                        default=False)
+
+    args = parser.parse_args()
+
+    return(args)
+
+def setup_logging(debug):
+    """Sets up the logging module and logging level."""
+
+    #Use capgen logging tools
+    logger = init_log('ccpp_track_variables')
+
+    if debug:
+        set_log_level(logger, logging.DEBUG)
+        logger.info('Logging level set to DEBUG')
+    else:
+        set_log_level(logger, logging.WARNING)
+    return logger
+
+def parse_suite(sdf, run_env):
+    """Reads the provided sdf, parses into a Suite data structure, including the "call tree":
+       the ordered list of schemes for the suite specified by the provided sdf"""
+    run_env.logger.info(f'Reading sdf {sdf} and populating Suite object')
+    suite = Suite(sdf_name=sdf)
+    run_env.logger.info(f'Reading sdf {sdf} and populating Suite object')
+    success = suite.parse(make_call_tree=True)
+    if not success:
+        raise Exception(f'Parsing suite definition file {sdf} failed.')
+    run_env.logger.info(f'Successfully read sdf {suite.sdf_name}')
+    return suite
+
+def create_metadata_filename_dict(metapath):
+    """Given a path, read all .meta files in that directory and add them to a dictionary: the keys
+       are the name of the scheme, and the values are the filename of the .meta file associated
+       with that scheme"""
+
+    metadata_dict = {}
+    scheme_filenames=glob.glob(os.path.join(metapath, "*.meta"))
+    if not scheme_filenames:
+        raise Exception(f'No files found in {metapath} with ".meta" extension')
+
+    for scheme_fn in scheme_filenames:
+        schemes=find_scheme_names(scheme_fn)
+        # The above returns a list of schemes in each filename, but
+        # we want a dictionary of schemes associated with filenames:
+        for scheme in schemes:
+            metadata_dict[scheme]=scheme_fn
+
+    return metadata_dict
+
+
+def create_var_graph(suite, var, config, metapath, run_env):
+    """Given a suite, variable name, a 'config' dictionary, and a path to .meta files:
+         1. Creates a dictionary associating schemes with their .meta files
+         2. Loops through the call tree of the provided suite
+         3. For each scheme, reads .meta file for said scheme, checks for variable within that
+            scheme, and if it exists, adds an entry to a list of tuples, where each tuple includes
+            the name of the scheme and the intent of the variable within that scheme"""
+
+    # Create a list of tuples that will hold the in/out information for each scheme
+    var_graph=[]
```

Comment:
Is there a pattern where you use spaces around the `=` symbol and where you omit them?

Reply:
Just regular sloppiness :) I've standardized this to use spaces except when in keyword arguments; I believe that's consistent with PEP8